### PR TITLE
Improve localization

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Fix a bunch of hard coded strings, see commits for specific bugs.

### CODE REVIEW

**Comment by amcasey** [comment]
>         public const string Name = "Node.js Interactive - Error";

Should these be private now?

---
Refers to: Nodejs/Product/InteractiveWindow/Repl/ReplOutputClassificationFormatDefinitions.cs:17 in b1291ee. [](commit_id = b1291ee535dcedacecb5b5f16ca79f3f5d5ea718, deletion_comment = False)

File: Nodejs/Product/InteractiveWindow/Repl/ReplWindow.cs [hunk 1]
```diff
@@ -2797,7 +2797,7 @@ private void ShowReplHelp()
             cmdnames.Sort((x, y) => StringComparer.CurrentCulture.Compare(x.Command, y.Command));
 
             const string helpFmt = "  {0,-24}  {1}";
-            WriteLine(string.Format(helpFmt, _commandPrefix + "help", "Show a list of REPL commands"));
+            WriteLine(string.Format(helpFmt, _commandPrefix + "help", Resources.helpDescription));
```
Comment (amcasey) [inline Nodejs/Product/InteractiveWindow/Repl/ReplWindow.cs]: Why lowercase?

File: Nodejs/Product/InteractiveWindow/Repl/Commands/EchoReplCommand.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@ public Task<ExecutionResult> Execute(IReplWindow window, string arguments)
             if (string.IsNullOrWhiteSpace(arguments))
             {
                 var curValue = (bool)window.GetOptionValue(ReplOptions.ShowOutput);
-                window.WriteLine("ECHO is " + (curValue ? "ON" : "OFF"));
+                window.WriteLine(string.Format(Resources.echoState, curValue ? "\'ON\'" : "\'OFF\'"));
```
Comment (amcasey) [inline Nodejs/Product/InteractiveWindow/Repl/Commands/EchoReplCommand.cs]: Format isn't built into WriteLine?

Comment (amcasey) [inline Nodejs/Product/InteractiveWindow/Repl/Commands/EchoReplCommand.cs]: Why wouldn't "On" and "Off" be localized?

File: Nodejs/Product/Nodejs/Resources.resx [hunk 3]
```diff
@@ -672,4 +672,40 @@ Error retrieving websocket debug proxy information from web.config.</value>
...
+    <value>Invalid filename: '{0}'.</value>
+  </data>
+  <data name="ReplSaveNoFileName" xml:space="preserve">
+    <value>Save requires a filename.</value>
+  </data>
+  <data name="ReplSaveSucces" xml:space="preserve">
+    <value>Session saved to: '{0}'.</value>
+  </data>
+  <data name="DebuggerAttachToDescription" xml:space="preserve">
+    <value>Allows attaching to Node.js processes running behind a remote debug proxy (RemoteDebug.js). Related documentation can be found under the 'Tools\Node.js Tool\Remote Debug Proxy' menu. Specify the target hostname and debugger port in the 'Qualifier' textbox, e.g. 'targethost: {0}'. This transport is not secure, and should not be used on a network that might have hostile traffic.</value>
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/Resources.resx]: >  [](start = 285, length = 1)

Extra space?

File: Nodejs/Product/Nodejs/Jade/Classifier/JadeClassificationTypes.cs [hunk 4]
```diff
@@ -1,8 +1,8 @@
-// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 namespace Microsoft.NodejsTools.Jade
 {
-    internal class JadeClassificationTypes
+    internal static class JadeClassificationTypes
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/Jade/Classifier/JadeClassificationTypes.cs]: >JadeClassificationTypes [](start = 26, length = 23)

Consider making these private to JadeClassificationDefinitions.

**Comment by amcasey** [comment]
>         [Name("JadeKeywordFormatDefinition")]

Why no constant?

---
Refers to: Nodejs/Product/Nodejs/Jade/Classifier/JadeClassificationDefinitions.cs:27 in b1291ee. [](commit_id = b1291ee535dcedacecb5b5f16ca79f3f5d5ea718, deletion_comment = False)

File: Nodejs/Product/Nodejs/Repl/SaveReplCommand.cs [hunk 5]
```diff
@@ -25,12 +25,12 @@ public Task<ExecutionResult> Execute(IReplWindow window, string arguments)
...
             if (string.IsNullOrWhiteSpace(arguments))
             {
-                window.WriteError("save requires a filename");
+                window.WriteError(Resources.ReplSaveNoFileName);
                 return ExecutionResult.Failed;
             }
             else if (arguments.IndexOfAny(Path.GetInvalidPathChars()) != -1)
             {
-                window.WriteError(string.Format(CultureInfo.CurrentCulture, "Invalid filename: {0}", arguments));
+                window.WriteError(string.Format(CultureInfo.CurrentCulture, Resources.ReplSaveInvalidFileName, arguments));
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/Repl/SaveReplCommand.cs]: >CurrentCulture [](start = 60, length = 14)

Not the `CurrentUICulture`?  Does it make a difference?

File: Nodejs/Product/InteractiveWindow/Repl/ReplWindow.cs (hunk 1 above)
Comment (paulvanbrenk) [reply Nodejs/Product/InteractiveWindow/Repl/ReplWindow.cs]: Copy/Paste

File: Nodejs/Product/InteractiveWindow/Repl/Commands/EchoReplCommand.cs (hunk 2 above)
Comment (paulvanbrenk) [reply Nodejs/Product/InteractiveWindow/Repl/Commands/EchoReplCommand.cs]: nope :(

Comment (paulvanbrenk) [reply Nodejs/Product/InteractiveWindow/Repl/Commands/EchoReplCommand.cs]: they are actual values you pass to the command, so keeping them as is seems less confusting

File: Nodejs/Product/Nodejs/Resources.resx (hunk 3 above)
Comment (paulvanbrenk) [reply Nodejs/Product/Nodejs/Resources.resx]: "generated by a tool"

File: Nodejs/Product/Nodejs/Jade/Classifier/JadeClassificationTypes.cs (hunk 4 above)
Comment (paulvanbrenk) [reply Nodejs/Product/Nodejs/Jade/Classifier/JadeClassificationTypes.cs]: They are used in other places, so no real benefit to make that change

File: Nodejs/Product/Nodejs/Repl/SaveReplCommand.cs (hunk 5 above)
Comment (paulvanbrenk) [reply Nodejs/Product/Nodejs/Repl/SaveReplCommand.cs]: Used throughout the solution, so not willing to risk that change right now

File: Nodejs/Product/InteractiveWindow/Repl/Commands/EchoReplCommand.cs (hunk 2 above)
Comment (amcasey) [reply Nodejs/Product/InteractiveWindow/Repl/Commands/EchoReplCommand.cs]: :(

File: Nodejs/Product/Nodejs/Resources.resx (hunk 3 above)
Comment (amcasey) [reply Nodejs/Product/Nodejs/Resources.resx]: What?

Comment (paulvanbrenk) [reply Nodejs/Product/Nodejs/Resources.resx]: aah

Comment (amcasey) [reply Nodejs/Product/Nodejs/Resources.resx]: I'm missing something.  There can't be a space between the colon and the port number, can there?

Comment (paulvanbrenk) [reply Nodejs/Product/Nodejs/Resources.resx]: ah yes, no. I'll fix that

**Pull request merged**